django/db/models/sql/compiler.py

```diff
@@ -157,7 +157,7 @@ def collapse_group_by(self, expressions, having):
                 }
                 expressions = [pk] + [
                     expr for expr in expressions
-                    if expr in having or getattr(expr, 'alias', None) not in pk_aliases
+                    if expr in having or (getattr(expr, 'alias', None) is not None and expr.alias not in pk_aliases)
```

Comment:
This is getting hard to read, I would recommend to name the separate parts, e.g.:
```
has_alias = getattr(expr, 'alias', None) is not None
```

Reply:
@codingjoe unless I'm missing something that would require stopping to use a list comprehension.

I guess the `alias` checking could be broken to a new line

```python
expressions = [pk] + [
    expr for expr in expressions
    if expr in having or (
        getattr(expr, 'alias', None) is not None and expr.alias not in pk_aliases
    )
]
```